builtin/pack/builder.go

```diff
@@ -68,6 +80,170 @@ var skipBuildPacks = map[string]struct{}{
 	"heroku/procfile": {},
 }
 
+// Build
+func (b *Builder) BuildODR(
+	ctx context.Context,
+	ui terminal.UI,
+	jobInfo *component.JobInfo,
+	src *component.Source,
+	log hclog.Logger,
+	ai *docker.AccessInfo,
+) (*DockerImage, error) {
+	sg := ui.StepGroup()
+
+	builder := b.config.Builder
+	if builder == "" {
+		builder = DefaultBuilder
+	}
+
+	log.Info("executing the ODR version of pack")
+
+	// We don't even need to inspect Docker to verify we have the image.
+	// If `pack` succeeded we can assume that it created an image for us.
+	// return &DockerImage{
+	// Image: ai.Image,
+	// Tag:   ai.Tag,
+	// }, nil
+
+	step := sg.Add("Building Buildpack with kaniko...")
+	defer func() {
+		if step != nil {
+			step.Abort()
+		}
+	}()
+
+	target := &docker.Image{
+		Image: ai.Image,
+		Tag:   ai.Tag,
+	}
+
+	var auth string
+
+	if ai.Auth != nil {
+		switch sv := ai.Auth.(type) {
+		case *docker.AccessInfo_Encoded:
+			user, pass, err := docker.CredentialsFromConfig(sv.Encoded)
+			if err != nil {
+				return nil, err
+			}
+			auth = ociregistry.BasicAuth(user, pass)
+		case *docker.AccessInfo_Header:
+			auth = sv.Header
+		}
+	}
+
+	// Determine the host that we're setting auth for. We have to parse the
+	// image for this cause it may not contain a host. Luckily Docker has
+	// libs to normalize this all for us.
+	log.Trace("determining host for auth configuration", "image", target.Name())
+	ref, err := reference.ParseNormalizedNamed(target.Name())
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "unable to parse image name: %s", err)
+	}
+	host := reference.Domain(ref)
+	log.Trace("auth host", "host", host)
+
+	var ocis ociregistry.Server
+	ocis.DisableEntrypoint = b.config.DisableCEB
+	ocis.Auth = auth
+	ocis.Logger = log
+	ocis.Upstream = "http://" + host
+
+	data, err := assets.Asset("ceb/ceb")
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "unable to restore custom entry point binary: %s", err)
+	}
+
+	refPath := reference.Path(ref)
+
+	step.Done()
+	step = sg.Add("Testing registry and uploading entrypoint layer")
+
+	err = ocis.SetupEntrypointLayer(refPath, data)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "error setting up entrypoint layer: %s", err)
+	}
+
+	li, err := net.Listen("tcp", ":5000")
```

Comment:
You might want to set this to `localhost:5000`. Otherwise I think this could bind to a non-loopback interface, and potentially be exposed on the internet.